receiver/splunkhecreceiver/factory.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecreceiver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configerror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+)
+
+// This file implements factory for Splunk HEC receiver.
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "splunk_hec"
+
+	// Default endpoints to bind to.
+	defaultEndpoint = ":8088"
+)
+
+// NewFactory creates a factory for SignalFx receiver.
+func NewFactory() component.ReceiverFactory {
+	return receiverhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiverhelper.WithMetrics(createMetricsReceiver))
+}
+
+// CreateDefaultConfig creates the default configuration for Splunk HEC receiver.
+func createDefaultConfig() configmodels.Receiver {
+	return &Config{
+		ReceiverSettings: configmodels.ReceiverSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+		Endpoint: defaultEndpoint,
+	}
+}
+
+// extract the port number from string in "address:port" format. If the
+// port number cannot be extracted returns an error.
+func extractPortFromEndpoint(endpoint string) (int, error) {
```

Comment:
Do you really need to do this?

Reply:
I lifted this from signalfxreceiver, looks like it's still there. Eventually something that could live in the common HTTPServerSettings struct.

Reply:
I don't understand why extracting the port vs using the endpoint as defined.

Reply:
It's just used for validation, apparently. This is something I copied from signalfxreceiver. No strong opinions: if you want it out, I can remove it.